tests/constraints/generic/test_main.py

```diff
@@ -18,6 +18,8 @@
         ("==win32", Constraint("win32", "=")),
         ("!=win32", Constraint("win32", "!=")),
         ("!= win32", Constraint("win32", "!=")),
+        ("not integra", Constraint("tegra", "not in")),
+        ("integra", Constraint("tegra", "in")),
```

Comment:
Yeah, this is how the `{op}{value}` is being generated. I was unsure about changing that method and introducing regression bugs with everything else. I'll play around with it to see if I can tweak it to increase readability without introducing many side effects

Reply:
Investigated a bit further, I can without much effort space out the operator from the value `not integra` -> `not in tegra`.

Changing the order of the values would require some reworking.
This one https://github.com/npapapietro/poetry-core/blob/ddfc68da2382e4581b6c4e6b278a7be56d584d44/src/poetry/core/constraints/generic/parser.py#L31 would require increasing the complexity of the regex itself or if statements to ensure mixing `not in`/`in` operators with others are still satisfied.

Reply:
Actually, I have no idea why this regex is so complicated. I think we can simplify it so it will not be an issue. See my other comment. I think I am still in favor of bringing it in the correct order despite some additional effort. Adapting `STR_CMP_CONSTRAINT` and `SingleMarker._CONSTRAINT_RE` should not be too complicated. In `STR_CMP_CONSTRAINT` we just have to swap `op` and `value` and in `SingleMarker._CONSTRAINT_RE` we can just add an "or" with swapped order for the new operators.